src/serac/physics/boundary_conditions/boundary_condition.hpp

```diff
@@ -133,140 +133,67 @@ class BoundaryCondition {
-   * @param[in] time The time for the coefficient, used for time-varying coefficients
-   */
-  void projectBdr(mfem::ParGridFunction& gf, const double time) const;
-
-  /**
-   * @brief Projects the boundary condition over boundary DOFs of a field
-   * @param[inout] state The field to project over
-   * @param[in] time The time for the coefficient, used for time-varying coefficients
-   */
-  void projectBdr(FiniteElementState& state, const double time) const;
-
-  /**
-   * @brief Projects the boundary condition over boundary DOFs
-   * @param[in] time The time for the coefficient, used for time-varying coefficients
-   * @pre A corresponding field (FiniteElementState) has been associated
-   * with the calling object via BoundaryCondition::setTrueDofs(FiniteElementState&)
-   */
-  void projectBdr(const double time) const;
-
-  /**
-   * @brief Projects the boundary condition over boundary to a DoF vector
-   * @param[in] dof_values The discrete dof values to project
-   * @param[in] time The time for the coefficient, used for time-varying coefficients
-   * @pre A corresponding field (FiniteElementState) has been associated
-   * with the calling object via BoundaryCondition::setTrueDofs(FiniteElementState&)
-   */
-  void projectBdrToDofs(mfem::Vector& dof_values, const double time) const;
-
-  /**
-   * @brief Eliminates the rows and columns corresponding to the BC's true DOFS
-   * from a stiffness matrix
-   * @param[inout] k_mat The stiffness matrix to eliminate from,
-   * will be modified.  These eliminated matrix entries can be
-   * used to eliminate an essential BC to an RHS vector with
-   * BoundaryCondition::eliminateToRHS
-   */
-  void eliminateFromMatrix(mfem::HypreParMatrix& k_mat) const;
-
-  /**
-   * @brief Eliminates boundary condition from solution to RHS
-   * @param[in] k_mat_post_elim A stiffness matrix post-elimination
-   * @param[in] soln The solution vector
-   * @param[out] rhs The RHS vector for the system
-   * @pre BoundaryCondition::eliminateFrom has been called
-   */
-  void eliminateToRHS(mfem::HypreParMatrix& k_mat_post_elim, const mfem::Vector& soln, mfem::Vector& rhs) const;
+  void setDofs(mfem::Vector& state, const double time = 0.0) const;
 
   /**
    * @brief Applies an essential boundary condition to RHS
-   * @param[in] k_mat_post_elim A stiffness (system) matrix post-elimination
+   * @param[inout] k_mat A stiffness (system) matrix post-elimination. The rows and cols of the essential dofs will be
+   * eliminated.
    * @param[out] rhs The RHS vector for the system
    * @param[inout] state The state from which the solution DOF values are extracted and used to eliminate
-   * @param[in] time Simulation time, used for time-varying boundary coefficients
-   * @pre BoundaryCondition::eliminateFrom has been called
+   * @pre The input state solution must contain the correct essential DOFs. This can be done by calling the @a
+   * BoundaryCondition::setDofs method.
    */
```

Comment:
Some ambiguities in this description:
- what does "elimination" mean here? This word typically means row-reduction in numerical linear algebra, but that's not what it means here.
- what is a "system" matrix?
- is `k_mat` required to be "post-elimination", or is this function the one to do the eliminating?
- is `state` an `inout` or just an input?

I'm not sure how mfem actually modifies the coefficient matrix (does it leave a 1 on the diagonal, or something else?), so the pseudocode below may be slightly inaccurate: 

```cpp
/**
 * @brief modify the system of equations, A x = b, by replacing equations that correspond to 
 * essential boundary conditions with ones that prescribe the desired values. To preserve symmetry,
 * the off-diagonal entries of associated columns of A are also zeroed out, and b is modified accordingly. 
 * This function is equivalent to the MATLAB code:
 * 
 *     A[essential_dofs] = eye(n)[essential_dofs];
 *     b -= A[:, essential_dofs] * x[essential_dofs];
 *     A[:, essential_dofs] = eye(n)[:, essential_dofs];
 * 
 * where `essential_dofs` are the rows corresponding to this essential boundary condition
 * 
 * @param[inout] A the coefficient matrix of the system of equations
 * @param[in] x the vector containing the prescribed values at the essential boundary condition indices 
 * @param[inout] b the righthand side
 *
 * @note `BoundaryCondition::setDofs` can be used to set the essential dofs of `x`.
 */

Reply:
Also: does this modify the essential dofs of `b`?

Reply:
Yes - it modifies the essential entries of `b`. I have modified the docs to reflect this.